### ISSUE TITLE
Add SYCL extension specification for max_work_group_size

### ISSUE THREAD

**Issue by steffenlarsen** [intel/llvm#7700]
DPC++ already supports `intel::max_work_group_size` but though the documentation for the attribute specifies the behavior at SYCL level, the attribute does not have a related SYCL extension detailing the behavior. Such an extensions should be added.

Additional options for this extension specification:

1. Allow 1-3 arguments in the attribute and add the requirement that the kernel must be launched with the same dimensionality as the attribute. This makes it in line with other SYCL 2020 core attributes for work groups (`sycl::reqd_work_group_size` and `sycl::work_group_size_hint`). See discussion https://github.com/intel/llvm/pull/7450#discussion_r1027925050.
2. Add new `sycl` spelling, i.e. `sycl::max_work_group_size`.
3. Add a kernel compile-time property analogue. See https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_oneapi_kernel_properties.asciidoc.
  ** Note: Since this would be experimental it may have to be in a separate extension.

**Comment by gmlueck** [comment]
> Add new `sycl` spelling, i.e. `sycl::max_work_group_size`.

We should not do this.  The `sycl` attribute namespace is reserved for core SYCL features, so we should not add extensions into this namespace.  See [section 6.3.4](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_attribute_namespace) "Attribute namespace".

Since we already support this attribute in the `intel` namespace, I think it would be best to just continue using that.  The long term direction is to move to compile-time properties.  When we do that, we should decide whether this property is in the `oneapi` vs. `intel` namespace.

**Comment by steffenlarsen** [comment]
> > Add new `sycl` spelling, i.e. `sycl::max_work_group_size`.
> 
> We should not do this. The `sycl` attribute namespace is reserved for core SYCL features, so we should not add extensions into this namespace. See [section 6.3.4](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_attribute_namespace) "Attribute namespace".
> 
> Since we already support this attribute in the `intel` namespace, I think it would be best to just continue using that. The long term direction is to move to compile-time properties. When we do that, we should decide whether this property is in the `oneapi` vs. `intel` namespace.

Good point. Whether or not we want it in the `intel` or `oneapi` attribute namespaces has an effect on the compile-time property naming however, so I think for the sake of consistency we should decide on one or the other before we add that.

**Comment by gmlueck** [comment]
Our general policy is to put extensions in the `oneapi` namespace unless they are specific to Intel hardware.  I presume this is not specific to Intel hardware, so it would go in the `oneapi` namespace?

Since you say that we already support `intel::max_work_group_size`, I presume we need to continue supporting this.  If we want to support a new attribute named `oneapi::max_work_group_size`, then we could mark `intel::max_work_group_size` as deprecated.

However, our long-term goal is to deprecate all these attributes and use compile-time properties instead.  Therefore, we would eventually deprecate `oneapi::max_work_group_size` also.

Rather than creating all this churn, it seems like it would be easier to just document `intel::max_work_group_size` for now.  When we are ready to implement the compile-time property (in the `oneapi` namespace), we can deprecate `intel::max_work_group_size`.  That seems like less churn to customers who are using it.  Is there a particular reason we need to implement the attribute in the `oneapi` namespace right away?
